File: scripts/process_backend/_02/pb02/java/ReadFile.java
import java.io.*;
import java.util.*;

public class ReadFile {
    public static void main(String[] args) throws Exception {
        String filename = args.length > 0 ? args[0] : "sample.csv";
        BufferedReader reader = new BufferedReader(new FileReader(filename));
        String line;
        List<String> lines = new ArrayList<>();

        long start = System.nanoTime();

        while ((line = reader.readLine()) != null) {
            lines.add(line);
        }

        long end = System.nanoTime();
        reader.close();

        System.out.printf("✅ Java read %d rows in %.4f seconds\n", lines.size(), (end - start) / 1e9);
    }
}

File: scripts/process_backend/_02/pb02/cpp/read_file.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <string>
#include <chrono>

int main(int argc, char* argv[]) {
    std::string filename = (argc > 1) ? argv[1] : "sample.csv";
    std::ifstream file(filename);
    std::vector<std::string> lines;
    std::string line;

    auto start = std::chrono::high_resolution_clock::now();

    while (std::getline(file, line)) {
        lines.push_back(line);
    }

    auto end = std::chrono::high_resolution_clock::now();
    std::chrono::duration<double> duration = end - start;

    std::cout << "✅ C++ read " << lines.size() << " rows in " << duration.count() << " seconds" << std::endl;
    return 0;
}